src/Xamarin.Android.Build.Tasks/Tasks/ResolveSdksTask.cs

```diff
@@ -302,6 +306,69 @@ public bool RunTask ()
 			return !Log.HasLoggedErrors;
 		}
 
+		static readonly Regex javaVersionRegex = new Regex (@"""([\d\.]+)_\d+""");
+
+		Version GetJavaVersionForFramework (string targetFrameworkVersion)
+		{
+			var apiLevel = AndroidVersion.TryOSVersionToApiLevel (targetFrameworkVersion);
+			if (apiLevel >= 24)
+				return new Version (1, 8);
+			else if (apiLevel == 23)
+				return new Version (1, 7);
+			else
+				return new Version (1, 6);
+		}
+
+		Version GetJavaVersionForBuildTools (string buildToolsVersion)
+		{
+			Version buildTools;
+			if (!Version.TryParse (buildToolsVersion, out buildTools)) {
+				return new Version (1, 7);
```

Comment:
Why default to Java 1.7 if we can't parse `buildToolsVersion`? Shouldn't we default to `LatestSupportedJavaVersion`?